cmd/erasure-utils.go

```
@@ -40,11 +41,19 @@ func newHashWriters(diskCount int, algo string) []hash.Hash {
 func newHash(algo string) hash.Hash {
 	switch algo {
 	case "blake2b":
-		return blake2b.New512()
+		h, err := blake2b.New512(nil)
+		if err != nil {
+			panic(fmt.Sprintf("impossible situation: %s", err))
```

Why is this returning an error? seems like all hashing `New()` never returns an error. any specific reason?

---

Also lets not do `fmt.Sprintf` here, we should be little more descriptive. 

```
panic(errUnexpected)
```

---

> Why is this returning an error? seems like all hashing New() never returns an error. any specific reason?

The blake2b implementation is based on [RFC 7693](https://tools.ietf.org/html/rfc7693) - blake2b can be used as a MAC, but the key must be between 0 and 64 bytes long - if it's longer, the New* functions will return an non-nil error.
But your right, the code should never return an error as long as no key is passed.

> Also lets not do fmt.Sprintf here, we should be little more descriptive.

Maybe something like: "used blake2b as MAC with an invalid key" or is ignoring the error the better behavior?

---

> Maybe something like: "used blake2b as MAC with an invalid key" or is ignoring the error the better behavior?

We can do this as well use blake2b.New512() from crypto if it returns an error , fall back to blake2b-simd project.

---

> We can do this as well use blake2b.New512() from crypto if it returns an error , fall back to blake2b-simd project. 

The line `h, err := blake2b.New512(nil)` can never fail. 
All `New` functions in the blake2b package only return an error, if a key is passed (this means: instead of using blake2b as hash function, use it as MAC, similar to HMAC)
But passing a key in this context (minio uses blake2b as hash function) would be a compatibility breaking change.

Maybe this example shows the implications:
```
// this never fails, because we use blake2b as a hash function
// theoretically the error can be ignored
h, err := blake2b.New512(nil)  

key := make([]byte, 32)
h, err := blake2b.New512(key)  // this line doesn't fail, but if somebody do:

key := make([]byte, 65)
h, err := blake2b.New512(key) // this line fails, because we use blake2b as MAC and the key is to long ( > 64 bytes)
```
I added the error checking just because I don't think completely ignoring errors is good style, but in this case it would be okay...
So blake2b-simd implementation would be in the repo, but never used...

---

Then i guess its fine.. we can ignore and add a comment why we are ignoring and why it wouldn't fail ever.